content/into.md

```diff
@@ -0,0 +1,319 @@
+
+TThe second scheme allows `into` as a soft modifier on traits, classes, and opaque type aliases. If a type definition is declared with this modifier, conversions to that type are allowed. The second scheme requires that one has control over the conversion target types so that an `into` can be added to their declaration. It is appropriate where there are a few designated types that are meant to be conversion targets. If that's the case, migration from Scala 2 to Scala 3 becomes easier since no function signatures need to be rewritten. We call this scheme "_into as a modifier_".
+
+
+## Motivation
+
+Scala 3's implicit conversions of the `scala.Conversion` class require a language import
+```
+import scala.language.implicitConversions
+```
+in any code that uses them as implicit conversions (code that calls conversions explicitly is not affected). If the import is missing, a feature warning is currently issued, and this will become an error in future versions of Scala 3. The motivation for this restriction is two-fold:
+
+ - Code with hidden implicit conversions is hard to understand and might have correctness or performance issues that go undetected.
+ - If we require explicit user opt-in for implicit conversions, we can significantly improve type inference by propagating expected type information more widely in those parts of the program where there is no opt-in.
+
+There is one broad use case, however, where implicit conversions are very hard to replace. This is the case where an implicit conversion is used to adapt a method argument to its formal parameter type. An example from the standard library:
+```scala
+scala> val xs = List(0, 1)
+scala> val ys = Array(2, 3)
+scala> xs ++ ys
+val res0: List[Int] = List(0, 1, 2, 3)
+```
+The input line `xs ++ ys` makes use of an implicit conversion from `Array[Int]` to `IterableOnce[Int]`. This conversion is defined in the standard library as an `implicit def`. Once the standard library is rewritten with Scala 3 conversions, this will require a language import at the use site, which is clearly unacceptable. It is possible to avoid the need for implicit conversions using method overloading or type classes, but this often leads to longer and more complicated code, and neither of these alternatives work for vararg parameters.
+
+## First Scheme: `into` as a Type Constructor
+
+This is where the `into` type constructor comes in. Here is a signature of a `++` method on `List[A]` that uses it:
+
+```scala
+  def ++ (elems: into[IterableOnce[A]]): List[A]
+```
+The `into` wrapper on the type of `elems` means that implicit conversions can be applied to convert the actual argument to an `IterableOnce` value, and this without needing a language import.
+
+`into` is defined as follows in the companion object of the `scala.Conversion` class:
+```scala
+opaque type into[T] >: T = T
```

Comment:
Commenting here to keep track of the loose idea of defining this as
```scala
opaque into type into[T] >: T = T
```
(not sure of the recommended order of `opaque` and `into` modifiers though).
Would that make any sense? Could that simplify the SIP by making the type `into` less special?

Reply:
@odersky any thought on this?

Reply:
I don't think there's enough of a match to pursue this.